kusto/test/etoe/etoe_test.go

```diff
@@ -581,6 +580,593 @@ func TestQueries(t *testing.T) {
 	}
 }
 
+func TestStatment(t *testing.T) {
+	t.Parallel()
+
+	if skipETOE || testing.Short() {
+		t.Skipf("end to end tests disabled: missing config.json file in etoe directory")
+	}
+
+	_, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	client, err := kusto.New(testConfig.kcsb)
+	if err != nil {
+		panic(err)
+	}
+
+	t.Cleanup(func() {
+		t.Log("Closing client")
+		require.NoError(t, client.Close())
+		t.Log("Closed client")
+	})
+
+	allDataTypesTable := fmt.Sprintf("goe2e_all_data_types_%d_%d", time.Now().UnixNano(), rand.Int())
+	require.NoError(t, createIngestionTable(t, client, allDataTypesTable, true))
+	dt, err := time.Parse(time.RFC3339Nano, "2020-03-04T14:05:01.3109965Z")
+	ts, err := time.ParseDuration("1h23m45.6789s")
+	guid, err := uuid.Parse("74be27de-1e4e-49d9-b579-fe0b331d3642")
+	tests := []struct {
+		// desc is a description of a test.
+		desc string
+		// stmt is the Kusot Stmt that will be sent.
+		stmt kusto.Statement
+		// setup is a function that will be called before the test runs.
+		setup func() error
+		// teardown is a functiont that will be called before the test ends.
+		teardown func() error
+		qcall    queryFunc
+		mcall    mgmtFunc
+		qjcall   queryJsonFunc
+		options  interface{} // either []kusto.QueryOption or []kusto.MgmtOption
+		// doer is called from within the function passed to RowIterator.Do(). It allows us to collect the data we receive.
+		doer func(row *table.Row, update interface{}) error
+		// gotInit creates the variable that will be used by doer's update argument.
+		gotInit func() interface{}
+		// want is the data we want to receive from the query.
+		want interface{}
+		// should the test fail
+		failFlag bool
+	}{
+		{
+			desc: "Complex query with Statement Builder - with String parameters",
```

Comment:
Nice, but you don't need to overload the server with 10 tests. Make one test with all of the checks (with ands).